File: packages/superwall/android/src/main/java/io/capawesome/capacitorjs/plugins/superwall/classes/results/GetSubscriptionStatusResult.java
package io.capawesome.capacitorjs.plugins.superwall.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.superwall.interfaces.Result;

public class GetSubscriptionStatusResult implements Result {

    @NonNull
    private final String status;

    public GetSubscriptionStatusResult(@NonNull String status) {
        this.status = status;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject jsObject = new JSObject();
        jsObject.put("status", status);
        return jsObject;
    }
}

File: packages/superwall/ios/Plugin/Classes/Results/GetSubscriptionStatusResult.swift
import Foundation
import Capacitor

@objc public class GetSubscriptionStatusResult: NSObject, Result {
    let status: String

    init(status: String) {
        self.status = status
    }

    @objc public func toJSObject() -> AnyObject {
        var jsResult = JSObject()
        jsResult["status"] = status
        return jsResult as AnyObject
    }
}
